packages/lib/src/components/Card/components/CardInput/useSRPanelForCardInputErrors.ts

```diff
@@ -0,0 +1,93 @@
+import { useEffect, useState } from 'preact/hooks';
+import useSRPanelContext from '../../../../core/Errors/useSRPanelContext';
+import { SetSRMessagesReturnFn } from '../../../../core/Errors/SRPanelProvider';
+import { handlePartialAddressMode, lookupBlurBasedErrors, mapFieldKey } from './utils';
+import { usePrevious } from '../../../../utils/hookUtils';
+import { SetSRMessagesReturnObject, SortedErrorObject } from '../../../../core/Errors/types';
+import { ERROR_ACTION_BLUR_SCENARIO, ERROR_ACTION_FOCUS_FIELD } from '../../../../core/Errors/constants';
+import { setFocusOnFirstField } from './handlers';
+import { getArrayDifferences } from '../../../../utils/arrayUtils';
+
+const useSRPanelForCardInputErrors = ({ errors, props, isValidating, retrieveLayout, specifications, billingAddress, sfp }) => {
+    // Extract fns from context
+    const { setSRMessagesFromObjects, setSRMessagesFromStrings, clearSRPanel, shouldMoveFocusSR } = useSRPanelContext();
+
+    // Generate a setSRMessages function - implemented as a partial, since the initial set of arguments don't change.
+    const setSRMessages: SetSRMessagesReturnFn = setSRMessagesFromObjects?.({
+        fieldTypeMappingFn: mapFieldKey
+    });
+
+    const partialAddressSchema = handlePartialAddressMode(props.billingAddressMode);
+
+    const [sortedErrorList, setSortedErrorList] = useState<SortedErrorObject[]>(null);
+
+    // Get the previous list of errors
+    const previousSortedErrors = usePrevious(sortedErrorList);
+
+    const sfStateErrorsObj = sfp.current?.mapErrorsToValidationRuleResult();
+    const mergedErrors = { ...errors, ...sfStateErrorsObj };
+
+    useEffect(() => {
```

Comment:
Maybe one improvement we could do is to wrap everything inside try catch block. So that any exceptions inside here won't block card rendering.

To demo we can try this code, it blocks rendering.
```js
    useEffect(() => {
            // the rest of the code
            throw new Error('dummy error');
    }, [errors]);
```
However, if we handle the error, it doesn't block rendering.
```js
    useEffect(() => {
        try {
            // all the code
            throw new Error('dummy error');
        } catch (e) {
// do nothing or send to analytics or log as warnings
            console.log({ e });
        }
    }, [errors]);
```

Reply:
I'l add it as a defensive-coding measure